sympy/polys/rings.py

```diff
@@ -2531,15 +2531,145 @@ def compose(f, x, a=None):
 
         return poly
 
+
+    def coeff_wrt(self, x, dg):
+        """
+        Coefficient of ``f`` with respect to ``x**deg``.
+
+        Treating ``f`` as a univariate polynomial in ``x`` this finds the
+        coefficient of ``x**deg`` as a polynomial in the other generators.
+
+        Parameters
+        ==========
+
+        p : sympy.Poly
+            The polynomial to compute the sparse coefficient for.
+        x : int
+            The symbol to compute the coefficient for.
+        dg : int
+            The degree of the monomial to compute the coefficient for.
+
+        Returns
+        =======
+
+        sympy.Expr
+            The expression of the polynomial ``f`` at the given symbol and degree.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> _, x, y, z = ring("x, y, z", ZZ)
+        >>> p = 2*x**4 + 3*y**4 + 10*z**2 + 10*x*z**2
+        >>> dg = 2
+        >>> x = 2
+        >>> p.coeff_wrt(x, dg)
+        10*x + 10
+
+        """
+        p = self
+        i = p.ring.index(x)
+        terms = [(m, c) for m, c in p.iterterms() if m[i] == dg]
+        monoms, coeffs = zip(*terms)
+        monoms = [m[:i] + (0,) + m[i + 1:] for m in monoms]
+        return p.ring.from_dict(dict(zip(monoms, coeffs)))
+
+
+    def prem(self, g, x=None):
+        """
+        Computes the pseudo-remainder of the polynomial `f` with respect to `g`.
+
+        The pseudo-remainder is a concept in polynomial division that provides a way to divide `f` by `g`
+        and obtain a remainder while still preserving some properties of division. It is defined as follows:
+
+        Given two polynomials `f` and `g`, the pseudo-remainder `r` of `f` with respect to `g` is a polynomial
+        such that there exist polynomials `q0`, `q1`, ..., `q(n-1)` satisfying the following conditions:
+
+        1. `f = q0 * g + r`
+        2. The degree of `r` is less than the degree of `g`
+        3. For each `i` from 0 to n-1, the degree of `q(i) * g` is less than or equal to the degree of `f`
+
+        Parameters
+        ==========
+
+        f : PolyElement
+            The polynomial to compute the pseudo-remainder for.
+        g : PolyElement
+            The polynomial to divide `f` by.
+        x : Symbol
+            The main variable of the polynomials.
+
+        Returns
+        =======
+
+        PolyElement: The pseudo-remainder polynomial.
+
+        Raises
+        ======
+
+        ZeroDivisionError : If the degree of `g` is negative or g is the zero polynomial.
+        ValueError : It is raised when the degree of `r` with respect to `x` stops decreasing and remains the same or increases in consecutive iterations of the while loop.
```

Comment:
We don't need to say "If the degree of `g` is negative or g is the zero polynomial" because these two things are equivalent. Only the zero polynomial has a negative degree. This should just say "if g is the zero polynomial".

I don't think that `ValueError` should be mentioned here. The impression I have is that it represents a condition that should never happen. I think the check is just there to pick up on a possible bug.